### PR TITLE
Fix actions in replies & janky updates

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-14437
affects: Student, Teacher
release note: none

### CODE REVIEW

**Comment by vanwagonet** [comment]
Test Plan:
twilson > student1 > Discussion Topics > DIscussions > Threaded (or any other place you can see deep threaded replies)
marking read/unread should work smoothly in deep threaded replies and in the root detail view
automatic read should work in both places
rating (liking) should animate smoothly now too
The author avatars should never blink during updates

File: Core/Core/Discussions/DiscussionHTML.swift [hunk 1]
```diff
@@ -0,0 +1,680 @@
...
+
+import Foundation
+
+enum DiscussionHTML {
+    // shortcuts to encode text for html
+    static func t(_ text: String?) -> String { CoreWebView.htmlString(text) }
+    static func s(_ text: String?) -> String { CoreWebView.jsString(text) }
+    static func s(_ styles: Styles) -> String { "'\(styles)'" }
+
+    // HTML string rendering for static content
```
Comment (vanwagonet) [inline Core/Core/Discussions/DiscussionHTML.swift:27]: I kept the static html rendering for use in compose reply & assignment details. I could make those use the same preact rendering, but it seemed wasteful to put a bunch of js into a loadHTMLString when it's going to be static thereafter.

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/xqor435) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fxqor435)
teacher: [Link](https://rebrand.ly/okzamwr) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fokzamwr)
parent: [Link](https://rebrand.ly/hpj98l3) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fhpj98l3)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/Core/Discussions/preact.min.js
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student, Teacher

[MBL-14437](https://instructure.atlassian.net/browse/MBL-14437)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.56% | 93.54% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**